ecommerce/models.py

```diff
@@ -415,10 +414,14 @@ class FulfillableOrder:
     """class to handle common logics like fulfill, enrollment etc"""
 
     def create_transaction(self, payment_data):
-        self.transactions.create(
-            data=payment_data,
-            amount=self.total_price_paid,
-        )
+        # Since transaction_id is unique identifier, only create transaction if transaction_id exists
+        transaction_id = payment_data.get("transaction_id")
+        if transaction_id:
```

Comment:
If `transaction_id` do we still want to continue with the enrollment of the user?  Are there any instances where `payment_data` does not contain the `transaction_id`?

Reply:
We may want to create some log output or something traceable when transaction_id is `None`.

Reply:
I am not sure if we need to block enrollment because of transaction_id from the API response.

There are only two types of transactions: payment and refund. This code is for payment, the refund is handled separately. For payment, my understanding is that it has to have transaction_id, otherwise it can't be recorded. This is also to block inserting some meaningless or manually added transaction data like I saw RC.

Yeah, I can add some log here if transaction_id is None 

 @rhysyngsun should we block the subsequent enrollment if no transaction_id? I haven't seen any payments that don't have transaction_id in them.

Reply:
Talked to @rachellougee about this and as long as the column is `NOT NULL` then that should protect us from null values since it'll correctly raise an exception. I think she was planning to wrap that call with some exception handling to make the error messaging a bit more specific though.